(staticcheck): file_anyvalue_proto_msgTypes[xxx].Exporter is deprecated

```
Error: protobufs/anyvalue.pb.go:441:3: file_anyvalue_proto_msgTypes[0].Exporter is deprecated: Exporter will be removed the next time we bump protoimpl.GenVersion. See https://github.com/golang/protobuf/issues/1640  (SA1019)
Error: protobufs/anyvalue.pb.go:453:3: file_anyvalue_proto_msgTypes[1].Exporter is deprecated: Exporter will be removed the next time we bump protoimpl.GenVersion. See https://github.com/golang/protobuf/issues/1640  (SA1019)
Error: protobufs/anyvalue.pb.go:465:3: file_anyvalue_proto_msgTypes[2].Exporter is deprecated: Exporter will be removed the next time we bump protoimpl.GenVersion. See https://github.com/golang/protobuf/issues/1[64](https://github.com/open-telemetry/opamp-go/actions/runs/11638143761/job/32412481855?pr=313#step:4:67)0  (SA1019)
Error: protobufs/anyvalue.pb.go:477:3: file_anyvalue_proto_msgTypes[3].Exporter is deprecated: Exporter will be removed the next time we bump protoimpl.GenVersion. See https://github.com/golang/protobuf/issues/1640  (SA1019)
Error: protobufs/opamp.pb.go:3685:3: file_opamp_proto_msgTypes[0].Exporter is deprecated: Exporter will be removed the next time we bump protoimpl.GenVersion. See https://github.com/golang/protobuf/issues/1640  (SA1019)
Error: protobufs/opamp.pb.go:3697:3: file_opamp_proto_msgTypes[1].Exporter is deprecated: Exporter will be removed the next time we bump protoimpl.GenVersion. See https://github.com/golang/protobuf/issues/1640  (SA1019)
Error: protobufs/opamp.pb.go:3709:3: file_opamp_proto_msgTypes[2].Exporter is deprecated: Exporter will be removed the next time we bump protoimpl.GenVersion. See https://github.com/golang/protobuf/issues/1640  (SA1019)
```

https://github.com/open-telemetry/opamp-go/actions/runs/11638143761/job/32412481855?pr=313#logs

--- a/.github/workflows/static-check.yml
+++ b/.github/workflows/static-check.yml
@@ -12,6 +12,9 @@ on:
 jobs:
   staticcheck:
     name: staticcheck (project)
+    strategy:
+      matrix:
+        dir: ["client", "internal", "server", "protobufshelpers"]
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v4
@@ -25,3 +28,4 @@ jobs:
         with:
           install-go: false
           version: "2023.1.7"
+          working-directory: ${{ matrix.dir }}

